components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -156,13 +140,25 @@ func (g *gpsrtk) receiveAndWriteCorrectionData() {
 	defer g.activeBackgroundWorkers.Done()
 	defer g.closeCorrectionWriter()
 
-	err := g.connectToNTRIP()
+	select {
+	case <-g.cancelCtx.Done():
+		return
+	default:
+	}
```

Comment:
what is this doing? just checking if the resource got closed early?

Reply:
Yup. It was originally at the top of `connectToNTRIP()`, which I inlined here. I'll remove it if you want: I doubt this would ever fire, and there's another one coming up on line 164 that does the same thing. It's here solely to not modify the old behavior.

Reply:
fine as is, but I guess I don't know the difference between a select statement checking Done() and an if statement checking ctx.Err()

Reply:
I think of them as equivalent, for this context. If there is a difference in performance, it's so small no one will notice. but you're right that this would be easier to read if it checked `.Err()` instead. I'll change it.

Reply:
...now that I look at it more, this is, like, the first thing done in a goroutine started in the constructor. There's no way the context is already canceled, and if it is, it won't be the end of the world to wait another few milliseconds to get down to the other check in the for loop. I'll remove this paragraph instead.

Reply:
...and that pushed me to simplify the exit conditions of this function further. It's looking way better than before; thanks for pushing me on this!